source/blog/_posts/2017-09-25-ajax-panel.md

```diff
@@ -0,0 +1,130 @@
+* list of entities
+* form to enter new data
+* list of related entities
+* some statistics
+
+Serving all this data from a single controller action may already be hard, now imagine the sheer number of parameters you may 
+have to exchange with this single action when you want to enable pagination, validation and search on this single page, 
+through this single action.
+
+This is pretty much the reason why stateless services and statefull frontends are a important topic in todays web development. 
+And it is exactly the reason why we created the ajax panel. It provides us with the means to load local state from the 
+server and create a rich ui.
+
+We evaluated different frameworks to achieve this target. AngularJS and Vue.JS were possible frameworks which allows
+two way data binding and statefull frontend access.
+
+<img src="/blog/img/ajax-panel-abstract.svg" alt="image">
+
+The small controller actions doesn't respond with the full page dom tree anymore. Each controller is only responsible for a
+specific panel content.
+
+Our target was also to use most of the already existing dependencies instead of adding new frameworks just for
+the B2B-Suite. As you maybe already know jQuery is a base dependency of Shopware 5.
+
+We decided to develop our own lightweight frontend framework on top of jQuery which allows asynchronous HTTP requests for our frontend.
+The main target of this framework is to execute the asynchronous call and render the response in a given and zoned DOM element by using event.
+The behaviour is very similar to angular's [zone.js](https://github.com/angular/zone.js).
+
+## Code Example
+The base structure of our ajax panel index action look like this:
+```html
+<div class="ajax-panel" data-url="http://domain.tld/ajax-panel-controller" data-id="example"></div>
+```
+
+On page load our jQuery ajax panel plugin will search for the class `ajax-panel` and use the data attribute `data-url`. 
+If this attribute contains a valid url the jQuery plugin will perform an asynchronous http request on the attribute url. 
+If the response is signed with a HTTP Status Code of 200 the response will be rendered back in the original zoned div with the data-id `example`.
+
+## Response Example
+
+The ajax panel controller action responds with the content of the ajax panel element and moves the dom structure inside the given element.
+The response could be like:
+
+```html
+<span>Example Ajax Content</span>
+```
+
+After the response is rendered in the parent Ajax Panel `div` the full dom structure will look like this:
+
+```html
+<div class="ajax-panel" data-url="http://domain.tld/ajax-panel-controller" data-id="example">
+
+    <span>Example Ajax Content</span>
+
+</div>
+```
+
+## Ajax Panel Plugins
+After we build our first views we run in several problems. The biggest problem was, that we want to use jQuery in the 
+response rendered content. We decided to develop an own plugin loader for our ajax panel which loads automatically
```

Comment:
"response rendered content"? Meinst du sowas wie "in the response's rendered content"? Weiß aber auch nicht, ob das richtig ist :)